### PR TITLE
Fix spotbugs warnings - Null pointer dereferences

### PR DESCRIPTION
**Pull request by mkemmerz**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Fixing Spotbugs warnings, mostly Null pointer dereferences and one minor typo. Unfortunately I didnt fix all of them, there are still 2 open Null pointer dereferences.

Not providing any tests but if it is wished I will do it.

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/plugins/thinbackup/restore/HudsonRestore.java [hunk 1]
```diff
@@ -193,13 +199,22 @@ private void restore(final File toRestore) throws IOException {
...
-
+    if (list == null) {
+      LOGGER
+           .severe("Cannot restore plugins because null is returned for files to restore.");
+      return;
+    }
     if (list.length != 1) {
       LOGGER
-          .severe("Cannot restore plugins because no or mulitble files with the name 'installedPlugins.xml' are in the backup.");
+          .severe("Cannot restore plugins because no or multiple files with the name 'installedPlugins.xml' are in the backup.");
```
Comment (StefanSpieker) [inline src/main/java/org/jvnet/hudson/plugins/thinbackup/restore/HudsonRestore.java:209]: nice catch.

File: src/main/java/org/jvnet/hudson/plugins/thinbackup/restore/HudsonRestore.java [hunk 2]
```diff
@@ -193,13 +199,22 @@ private void restore(final File toRestore) throws IOException {
...
       return;
     }
 
+    if (list[0] == null) {
+      LOGGER
```
Comment (StefanSpieker) [inline src/main/java/org/jvnet/hudson/plugins/thinbackup/restore/HudsonRestore.java:214]: another linebreak?

File: src/main/java/org/jvnet/hudson/plugins/thinbackup/backup/HudsonBackup.java [hunk 3]
```diff
@@ -371,10 +371,10 @@ private void backupNextBuildNumberFile(final File jobDirectory, final File jobBa
...
     if (plugin.isBackupBuildResults()) {
       final File buildsDir = new File(jobDirectory, BUILDS_DIR_NAME);
-      if (buildsDir.exists() && buildsDir.isDirectory()) {
-        final Collection<String> builds = Arrays.asList(buildsDir.list());
-        if (builds != null) {
-          TopLevelItem job = hudson.getItem(jobDirectory.getName());
+      if (buildsDir.list() != null && buildsDir.exists() && buildsDir.isDirectory()) {
+        final String[] builds = buildsDir.list();
+        TopLevelItem job = hudson.getItem(jobDirectory.getName());
+        if (builds != null)
```
Comment (StefanSpieker) [inline src/main/java/org/jvnet/hudson/plugins/thinbackup/backup/HudsonBackup.java:377]: Please re-add the curly bracket for the if.

**Review by StefanSpieker — COMMENTED** [review]
Oterwise it looks quite good. Thanks a lot.

**Comment by mkemmerz** [comment]
I added your suggestions. Removed the linebreaks and added the bracket

**Review by StefanSpieker — APPROVED** [review]

**Pull request merged**